updater/lib/dependabot/updater/update_helpers.rb

```diff
@@ -0,0 +1,97 @@
+# typed: true
+# frozen_string_literal: true
+
+require "sorbet-runtime"
+
+# This module extracts all helpers required to perform additional update job
+# error recording and logging for various updates since they are shared
+# between a few operations.
+module Dependabot
+  class Updater
+    module UpdateHelpers
+      extend T::Sig
+      extend T::Helpers
+
+      abstract!
+
+      sig { returns(Dependabot::Service) }
+      attr_reader :service
+
+      # Logs a deprecation warning for a eco system version that is deprecated but the job will still succeed.
+      #
+      # @param eco_system [String] the name of the eco system
+      # @param deprecated_version [String] the deprecated version of the eco system
+      # @param supported_versions [Array<String>] the versions of the eco system that are still supported
+      #
+      # @example
+      #   record_deprecation_warning_for_eco_system(
+      #     eco_system: "bundler",
+      #     deprecated_version: "v1",
+      #     supported_versions: ["v2"]
+      #   )
+      sig do
+        params(
+          eco_system: String,
+          deprecated_version: String,
+          supported_versions: T::Array[String]
+        ).void
+      end
+      def record_deprecation_warning_for_eco_system(
```

Comment:
wondering if we could pass the mode as an enum? E.g. as `WARN` or `ERROR`; that way we can reuse the same function.

Reply:
I will check that and I think it is possible and depends to how it is used it may be better solution.

Reply:
The code is moved https://github.com/dependabot/dependabot-core/pull/10466